metrics/deny_list.go

```diff
@@ -0,0 +1,70 @@
+// Copyright 2021 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package metrics
+
+import (
+	"regexp"
+)
+
+// DenyList encapsulates the logic needed to filter based on a string.
+type DenyList struct {
```

Comment:
You must provide unit tests for this struct.

Reply:
Unit tests are still missing as far as I can see.

Reply:
I will provide unit tests as soon as possible.

Reply:
Hi, iwankgb,I have provided unit tests.